aws/adapter.go

```diff
@@ -67,24 +74,42 @@ func defaultConfigProvider() client.ConfigProvider {
 // Before returning there is a discovery process for VPC and EC2 details. It tries to find the TargetGroup and
 // Security Group that should be used for newly created LoadBalancers. If any of those critical steps fail
 // an appropriate error is returned.
-func NewAdapter(healthCheckPath string, healthCheckPort uint16) (adapter *Adapter, err error) {
+func NewAdapter(healthCheckPath string, healthCheckPort uint16, certUpdateInterval time.Duration) (adapter *Adapter, err error) {
 	p := configProvider()
 	adapter = &Adapter{
 		elbv2:           elbv2.New(p),
 		ec2:             ec2.New(p),
 		ec2metadata:     ec2metadata.New(p),
 		autoscaling:     autoscaling.New(p),
+		acm:             acm.New(p),
+		cc:              nil,
 		healthCheckPath: healthCheckPath,
 		healthCheckPort: healthCheckPort,
 	}
+	adapter.cc = adapter.NewAcm(certUpdateInterval)
 
 	adapter.manifest, err = buildManifest(adapter)
 	if err != nil {
 		return nil, err
 	}
+
 	return
 }
 
+func (a *Adapter) NewAcm(certUpdateInterval time.Duration) *certificateCache {
+	cc := NewCertCache(a.acm)
+	cc.InitCertCache(certUpdateInterval)
+	return cc
+}
+
+func (a *Adapter) GetCerts() []*acm.CertificateDetail {
```

Comment:
The adapter shouldn't export any AWS SDK specific types. Create a wrapper struct